testutil/testutil.go

```diff
@@ -0,0 +1,98 @@
+package testutil
+
+import (
+	"net"
+	"sync"
+)
+
+// Test constants
+const (
+	CloseOnMsgIdx = 5
+	MaxMsgCount   = 10
+)
+
+// LocalTCPServer tcp server for testing specific network errors
+type LocalTCPServer struct {
+	lnmu     sync.RWMutex
+	MockConn *Conn
```

Comment:
I agree. If we can force specific connection errors (i.e. timeouts/resets/etc) then we can abandon this approach for sure. I'm playing with some stuff @mattaitchison sent over to see if we can use a "normal" `net.Conn` object.